analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.cs

```diff
@@ -408,7 +408,63 @@ void ShortCircuitOrRight(int arg)
             }
             else
             {
-                o.ToString();   // Compliant
+                o.ToString();   // Compliant, unreachable
+            }
+        }
+
+        void BinaryAndLeft(int arg)
+        {
+            object o = null;
+            var isFalse = false;
+            if (isFalse & arg == 0)
+            {
+                o.ToString();   // Compliant, unreachable
+            }
+            else
+            {
+                o.ToString();   // Noncompliant
+            }
+        }
+
+        void BinaryAndRight(int arg)
+        {
+            object o = null;
+            var isFalse = false;
+            if (arg == 0 & isFalse)
+            {
+                o.ToString();   // Compliant, unreachable
+            }
+            else
+            {
+                o.ToString();   // Noncompliant
+            }
+        }
+
+        void ShortCircuitAndLeft(int arg)
+        {
+            object o = null;
+            var isFalse = false;
+            if (isFalse && arg == 0)
```

Comment:
Do we have a test for this case already?
```cs
if (isFalse && o.ToString()) // Compliant o.ToString is unreachable
{
}
```
If not add it.

Reply:
`o.ToString()` doesn't convert to bool.

Reply:
Added `if (isFalse && o.ToString() == null)`